pkg/controllers/controllers.go

```diff
@@ -113,9 +113,11 @@ func Initialize(injectCloudProvider func(context.Context, cloudprovider.Options)
 		utilruntime.Must(registerPprof(manager))
 	}
 
-	cloudProvider := cloudprovidermetrics.Decorate(
-		injectCloudProvider(ctx, cloudprovider.Options{ClientSet: clientSet, KubeClient: manager.GetClient(), StartAsync: manager.Elected()}),
-	)
+	cloudProvider := injectCloudProvider(ctx, cloudprovider.Options{ClientSet: clientSet, KubeClient: manager.GetClient(), StartAsync: manager.Elected()})
+	if hp, ok := cloudProvider.(HealthCheck); ok {
```

Comment:
Another option is to write an AWS specific controller that checks the liveness of the cloudprovider. I've been wanting to build a simple mechanism to wire additional controllers through Initialize()

Reply:
I was hoping to make these health checks better when I started this, but couldn't really come up with much apart from preventing deadlock.  If we can come up with any scenario where restarting is the best corrective action, we can just hook it up to a health check and return it as an error which will provide a bit more info on why we restarted.